connd/qconnectionagent.cpp

```diff
@@ -462,19 +481,25 @@ void QConnectionAgent::techChanged()
         tetheringWifiTech = 0;
         return;
     }
-    if (tetheringWifiTech) {
-        tetheringEnabled = tetheringWifiTech->tethering();
+    if (!netman->getTechnology("bluetooth")) {
+        tetheringBtTech = 0;
+        return;
     }
 
-    Q_FOREACH(NetworkTechnology *technology,netman->getTechnologies()) {
+    Q_FOREACH(NetworkTechnology *technology, netman->getTechnologies()) {
```

Comment:
```suggestion
    for (NetworkTechnology *technology : netman->getTechnologies()) {
```

If doing style fix along the lines please change Q_FOREACH to C++ range-based for. Usage for Q_FOREACH discouraged since 5.7.

Reply:
I'll replace all of these.